pkg/decoder/decoder.go

```diff
@@ -28,8 +28,8 @@ func (d *decoder) uint8() uint8     { return d.buf.Next(1)[0] }
 func (d *decoder) uint16() uint16   { return d.order.Uint16(d.buf.Next(2)) }
 func (d *decoder) uint32() uint32   { return d.order.Uint32(d.buf.Next(4)) }
 func (d *decoder) uint64() uint64   { return d.order.Uint64(d.buf.Next(8)) }
-func (d *decoder) OID() dbutils.OID { return dbutils.OID(d.uint32()) }
-func (d *decoder) LSN() dbutils.LSN { return dbutils.LSN(d.uint64()) }
+func (d *decoder) oid() dbutils.OID { return dbutils.OID(d.uint32()) }
```

Comment:
Go visibility rules interfering with our naming convention; another argument for renaming those functions to decodeNextValueAs$Type or something along this line.